<!--
  1 failure:  :disappointed_rel...
  1 warning:  There is no linke...
  1 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a>
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  
> titanium-mobile@7.1.0 test /Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9655-WMZXIQ33ORU4OOJZFL3OSRMYL4M2A52BZBB35JJTZHO54CVXX5VQ
> grunt

Running "appcJs:src" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9655-WMZXIQ33ORU4OOJZFL3OSRMYL4M2A52BZBB35JJTZHO54CVXX5VQ/android/cli/commands/_build.js
    17:1  error    Use the global form of 'use strict'  strict
  1634:9  warning  Blocks are nested too deeply (6)     max-depth

✖ 2 problems (1 error, 1 warning)

Warning: Task "eslint:src" failed. Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -147,11 +147,18 @@ timestamps {
 						// FIXME Do we need to do anything special to make sure we get os-specific modules only on that OS's build/zip?
 						sh 'npm install'
 					}
-					// Stash files for danger.js later
-					if (isPR) {
-						stash includes: 'node_modules/,package.json,package-lock.json,dangerfile.js', name: 'danger'
+					// Run npm test, but record output in a file and check for failure of command by checking output
+					def npmTestResult = sh(returnStatus: true, script: 'npm test &> npm_test.log')
+					if (isPR) { // Stash files for danger.js later
+						stash includes: 'node_modules/,package.json,package-lock.json,dangerfile.js,npm_test.log', name: 'danger'
 					}
-					sh 'npm test' // Run linting first // TODO Record the eslint output somewhere for danger to use later?
+					// was it a failure?
+					if (npmTestResult != 0) {
+						// empty stashes of test reports, so danger step can still run.
+						stash allowEmpty: true, name: 'test-report-ios'
+						stash allowEmpty: true, name: 'test-report-android'
+						error readFile('npm_test.log')
+					}
 				}
 
 				// Skip the Windows SDK portion if a PR, we don't need it

--- a/dangerfile.js
+++ b/dangerfile.js
@@ -20,6 +20,15 @@ if (github.pr.deletions > github.pr.additions) {
 	message(':thumbsup: Hey!, You deleted more code than you added. That\'s awesome!');
 }
 
+// Check npm test output
+if (fs.existsSync('./npm_test.log')) {
+	const npmTestOutput = fs.readFileSync('./npm_test.log');
+	if (npmTestOutput.indexOf('Test failed.  See above for more details.') !== -1) {
+		fail(':disappointed_relieved: `npm test` failed. See below for details.');
+		message('```' + npmTestOutput + '\n```');
+	}
+}
+
 // TODO Check for PRs above a certain threshold of changes and warn?
 
 // Check that we have a JIRA Link in the body

